xdsl/dialects/omp.py

```diff
@@ -164,6 +281,191 @@ class TerminatorOp(IRDLOperation):
     traits = traits_def(IsTerminator())
 
 
+@irdl_op_definition
+class TargetOp(IRDLOperation):
+    """
+    Implementation of upstream omp.target
+    https://mlir.llvm.org/docs/Dialects/OpenMPDialect/ODS/#omptarget-omptargetop
+    """
+
+    name = "omp.target"
+
+    allocate_vars = var_operand_def()
+    allocator_vars = var_operand_def()
+    depend_vars = var_operand_def()  # TODO: OpenMP_PointerLikeTypeInterface
+    device = opt_operand_def(IntegerType | IndexType)
+    has_device_addr_vars = var_operand_def()  # TODO: OpenMP_PointerLikeTypeInterface
+    host_eval_vars = var_operand_def()
+    if_expr = opt_operand_def(i1)
+    in_reduction_vars = var_operand_def()  # TODO: OpenMP_PointerLikeTypeInterface
+    is_device_ptr_vars = var_operand_def()  # TODO: OpenMP_PointerLikeTypeInterface
+    map_vars = var_operand_def()  # TODO: OpenMP_PointerLikeTypeInterface
+    private_vars = var_operand_def()
+    thread_limit = opt_operand_def(IntegerType | IndexType)
+
+    in_reduction_syms = opt_prop_def(ArrayAttr[SymbolRefAttr])
+    nowait = opt_prop_def(UnitAttr)
+    private_syms = opt_prop_def(ArrayAttr[SymbolRefAttr])
```

Comment:
Is there a reason to omit the remaining properties?

Reply:
I think the reason was to not add a lot of new attributes which these properties required.

I can pull them in too, but that would increase the size of the PR even further